hugo/content/docs/dependency_injection.md

```diff
@@ -0,0 +1,91 @@
+
+## Why dependency injection
+DI allows us to
+- not care about the instantiation and life-cycle of service components,
+- manage singletons like the various registries without using the global scope,
+- easily mock components in tests,
+- exchange default implementations with custom ones with minimum code changes,
+- modularize the configuration of specific features and scenarios and merge these modules for the final application
+
+## The container
+The DI-container is the main point of configuration. The standard in sprotty is naming the file containing the container `di.config.ts`.
+
+```typescript
+export const createContainer = (containerId: string) => {
+    const myModule = new ContainerModule((bind, unbind, isBound, rebind) => {
+        bind(TYPES.ModelSource).to(LocalModelSource).inSingletonScope();
+        const context = { bind, unbind, isBound, rebind };
+        configureModelElement(context, 'graph', SGraph, SGraphView);
+        configureModelElement(context, 'task', SNode, TaskNodeView);
+        configureModelElement(context, 'edge', SEdge, PolylineEdgeView);
+        configureViewerOptions(context, {
+            needsClientLayout: false,
+            baseDiv: containerId
+        });
+    });
+    const container = new Container();
+    loadDefaultModules(container);
+    container.load(myModule, edgeIntersectionModule);
+    return container;
+};
+```
+The container is build from multiple modules. Through `loadDefaultModules()` all modules are loaded for default sprotty functionality. we can of course also load other non default modules like the `edgeIntersectionModule` for extra functionality.
+
+Most important is our own module where the core of the configuration happens. Here we can configure singleton scope classes like our [datasource](../datasources) or rebind default default sprotty components like for example the logger to a custom implementation. We use Symbols for bindings instead of using classes directly. All Symbols can be found in the `TYPES` object.
+
+Through `configureModelElement` we can link our model to specific view components through the type property. Meaning if we have 
+```Typescript
+<SNode & TaskNode>{
+            type: 'task',
+            id: 'task01',
+            name: 'First Task',
+            isFinished: true,
+            ...
+}
+```
+in our model, sprotty will try to convert this datastructure to an instance of the actual `SNode` class and render it with the `TaskNodeView`.
+
+lastly we configure our viewer options. Here we configure all the dom elements needed by sprotty like for example the base div where our Diagram is rendered or the hidden div used by the first render cycle for determining micro layout. Another thing configured here is layouting. Specificily should layouting be done on client or servreside or both. This also determines the protocol spoken by client and server. 
```

Comment:
```suggestion
Lastly, we configure our viewer options. Here we configure all the dom elements needed by Sprotty, for example the base div inside of which our diagram is rendered or the hidden div used by the first render cycle for determining micro layout. Another thing configured here is layouting. Specifically if layouting should be done on client-side, server-side, or both. This also determines the protocol spoken by client and server. 
```

It could be good to have an example of those configs in the code above